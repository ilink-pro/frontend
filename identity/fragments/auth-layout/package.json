{
  "name": "@identity/auth-layout-fragment",
  "version": "0.0.1",
  "license": "BSD-3-Clause",
  "main": "src/index.ts",
  "devDependencies": {
    "@emotion/react": "^11.9.0",
    "@emotion/styled": "^11.8.1",
    "@testing-library/react": "^12.1.2",
    "@types/react": "^17.0.8",
    "@types/styled-system": "^5.1.11",
    "@ui/layout": "workspace:0.0.1",
    "@ui/text": "workspace:0.0.1",
    "@ui/theme": "workspace:0.0.1",
    "react": "^17.0.2",
    "react-dom": "^17.0.2",
    "react-intl": "^6.0.2",
    "styled-system": "^5.1.5",
    "styled-tools": "^1.7.2"
  },
  "peerDependencies": {
    "react": "^17.0.1",
    "react-dom": "^17.0.2"
  }
}
