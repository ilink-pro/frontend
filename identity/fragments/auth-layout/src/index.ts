export * from './auth-layout.component'
export * from './auth-layout.interfaces'
