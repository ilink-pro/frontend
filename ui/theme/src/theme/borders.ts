const borders = {
  soft: '1px solid #E3DEF0',
}

export { borders }
