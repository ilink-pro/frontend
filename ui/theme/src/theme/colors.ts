const colors = {
  background: {
    white: 'rgb(255,255,255)',
    lightPurple: 'rgb(251,250,255)',
    lightPurpleAccent: 'rgb(219,216,238)',
    purpleAccent: 'rgb(116,97,240)',
  },
  text: {
    primary: 'rgb(54,52,61)',
    accent: 'rgb(116,97,240)',
    secondary: 'rgb(153,147,189)',
    red: 'rgb(220,103,103)',
    gray: 'rgb(155,164,167)',
    white: 'rgb(255,255,255)',
  },
  button: {
    secondary: {
      default: {
        background: 'rgb(255,255,255)',
        border: 'rgb(219,216,238)',
        font: 'rgb(116,97,240)',
      },
      hover: {
        background: 'rgb(255,255,255)',
        border: 'rgb(116,97,240)',
        font: 'rgb(116,97,240)',
      },
      pressed: {
        background: 'rgb(255,255,255)',
        border: 'rgb(231,228,251)',
        font: 'rgb(116,97,240)',
      },
      disabled: {
        background: 'rgb(255,255,255)',
        border: 'rgb(219,216,238)',
        font: 'rgb(153,147,189)',
      },
    },
  },
  select: {
    menu: {
      border: 'rgb(219,216,238)',
    },
    item: {
      default: {
        background: 'rgb(255,255,255)',
      },
      hover: {
        background: 'rgb(247,246,254)',
      },
    },
  },
}

export { colors }
