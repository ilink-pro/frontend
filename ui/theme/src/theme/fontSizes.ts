export const fontSizes = {
  small: 12,
  semiRegular: 14,
  default: 16,
  increased: 18,
  semiGiant: 24,
  semiBig: 28,
  huge: 32,
  big: 36,
  giant: 40,
  colossal: 64,
}
