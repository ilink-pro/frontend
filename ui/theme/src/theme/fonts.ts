const fonts = {
  primary: 'Khula',
}

export { fonts }
