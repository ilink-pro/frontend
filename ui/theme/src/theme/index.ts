export * from './fontWeights'
export * from './lineHeights'
export * from './fontSizes'
export * from './colors'
export * from './fonts'
export * from './radii'
export * from './shadows'
export * from './borders'
