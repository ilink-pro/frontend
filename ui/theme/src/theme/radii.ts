const radii = {
  normal: 8,
  semiMedium: 12,
  medium: 16,
  full: '100%',
}

export { radii }
