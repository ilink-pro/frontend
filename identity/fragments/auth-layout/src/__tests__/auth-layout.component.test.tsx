/**
 * @jest-environment jsdom
 */

import { render }        from '@testing-library/react'

import React             from 'react'
import { IntlProvider }  from 'react-intl'

import { ThemeProvider } from '@ui/theme'

import { AuthLayout }    from '../auth-layout.component'

describe('identity', () => {
  describe('auth-layout', () => {
    describe('auth-layout.component', () => {
      it('should match the latest render snapshot', () => {
        const { asFragment } = render(
          <IntlProvider locale='ru' messages={{}}>
            <ThemeProvider>
              <AuthLayout title='title' />
            </ThemeProvider>
          </IntlProvider>
        )

        expect(asFragment()).toMatchSnapshot()
      })
    })
  })
})
