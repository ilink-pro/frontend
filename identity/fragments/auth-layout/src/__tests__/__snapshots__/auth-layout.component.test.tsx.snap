// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`identity auth-layout auth-layout.component should match the latest render snapshot 1`] = `
<DocumentFragment>
  .emotion-0 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  background-color: rgb(251,250,255);
  width: 100%;
  height: 100%;
}

.emotion-1 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-basis: 0;
  -ms-flex-preferred-size: 0;
  flex-basis: 0;
}

@media screen and (min-width: 40em) {
  .emotion-1 {
    -webkit-flex-basis: 0;
    -ms-flex-preferred-size: 0;
    flex-basis: 0;
  }
}

@media screen and (min-width: 52em) {
  .emotion-1 {
    -webkit-flex-basis: 160px;
    -ms-flex-preferred-size: 160px;
    flex-basis: 160px;
  }
}

.emotion-2 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 100%;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.emotion-3 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-basis: 0;
  -ms-flex-preferred-size: 0;
  flex-basis: 0;
}

@media screen and (min-width: 40em) {
  .emotion-3 {
    -webkit-flex-basis: 0;
    -ms-flex-preferred-size: 0;
    flex-basis: 0;
  }
}

@media screen and (min-width: 52em) {
  .emotion-3 {
    -webkit-flex-basis: 64px;
    -ms-flex-preferred-size: 64px;
    flex-basis: 64px;
  }
}

.emotion-4 {
  box-sizing: border-box;
  width: 100%;
  height: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  background-color: rgb(255,255,255);
  box-shadow: 0px 7px 16px rgba(136, 165, 191, 0.12);
  border-radius: 12px;
}

@media screen and (min-width: 40em) {
  .emotion-4 {
    width: 100%;
    height: 100%;
  }
}

@media screen and (min-width: 52em) {
  .emotion-4 {
    width: 672px;
    height: auto;
  }
}

.emotion-5 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-basis: 16px;
  -ms-flex-preferred-size: 16px;
  flex-basis: 16px;
}

@media screen and (min-width: 40em) {
  .emotion-5 {
    -webkit-flex-basis: 16px;
    -ms-flex-preferred-size: 16px;
    flex-basis: 16px;
  }
}

@media screen and (min-width: 52em) {
  .emotion-5 {
    -webkit-flex-basis: 40px;
    -ms-flex-preferred-size: 40px;
    flex-basis: 40px;
  }
}

.emotion-6 {
  box-sizing: border-box;
  width: 343px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

@media screen and (min-width: 40em) {
  .emotion-6 {
    width: 343px;
  }
}

@media screen and (min-width: 52em) {
  .emotion-6 {
    width: 504px;
  }
}

.emotion-7 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 100%;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 100%;
  height: 100%;
}

.emotion-8 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-basis: 48px;
  -ms-flex-preferred-size: 48px;
  flex-basis: 48px;
}

@media screen and (min-width: 40em) {
  .emotion-8 {
    -webkit-flex-basis: 48px;
    -ms-flex-preferred-size: 48px;
    flex-basis: 48px;
  }
}

@media screen and (min-width: 52em) {
  .emotion-8 {
    -webkit-flex-basis: 40px;
    -ms-flex-preferred-size: 40px;
    flex-basis: 40px;
  }
}

.emotion-9 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.emotion-10 {
  color: rgb(54,52,61);
  font-size: 28px;
  font-weight: 700;
  font-family: Khula;
  line-height: 1;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
}

@media screen and (min-width: 40em) {
  .emotion-10 {
    font-size: 28px;
  }
}

@media screen and (min-width: 52em) {
  .emotion-10 {
    font-size: 36px;
  }
}

.emotion-11 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-basis: 16px;
  -ms-flex-preferred-size: 16px;
  flex-basis: 16px;
}

@media screen and (min-width: 40em) {
  .emotion-11 {
    -webkit-flex-basis: 16px;
    -ms-flex-preferred-size: 16px;
    flex-basis: 16px;
  }
}

@media screen and (min-width: 52em) {
  .emotion-11 {
    -webkit-flex-basis: 12px;
    -ms-flex-preferred-size: 12px;
    flex-basis: 12px;
  }
}

.emotion-13 {
  color: rgb(155,164,167);
  font-size: 16px;
  font-family: Khula;
  font-weight: 400;
  line-height: 1;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
}

.emotion-14 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-basis: 48px;
  -ms-flex-preferred-size: 48px;
  flex-basis: 48px;
}

@media screen and (min-width: 40em) {
  .emotion-14 {
    -webkit-flex-basis: 48px;
    -ms-flex-preferred-size: 48px;
    flex-basis: 48px;
  }
}

@media screen and (min-width: 52em) {
  .emotion-14 {
    -webkit-flex-basis: 32px;
    -ms-flex-preferred-size: 32px;
    flex-basis: 32px;
  }
}

.emotion-15 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
  height: 100%;
}

.emotion-16 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-basis: 16px;
  -ms-flex-preferred-size: 16px;
  flex-basis: 16px;
}

@media screen and (min-width: 40em) {
  .emotion-16 {
    -webkit-flex-basis: 16px;
    -ms-flex-preferred-size: 16px;
    flex-basis: 16px;
  }
}

@media screen and (min-width: 52em) {
  .emotion-16 {
    -webkit-flex-basis: 128px;
    -ms-flex-preferred-size: 128px;
    flex-basis: 128px;
  }
}

.emotion-17 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-basis: 0;
  -ms-flex-preferred-size: 0;
  flex-basis: 0;
}

@media screen and (min-width: 40em) {
  .emotion-17 {
    -webkit-flex-basis: 0;
    -ms-flex-preferred-size: 0;
    flex-basis: 0;
  }
}

@media screen and (min-width: 52em) {
  .emotion-17 {
    -webkit-flex-basis: 128px;
    -ms-flex-preferred-size: 128px;
    flex-basis: 128px;
  }
}

<div
    class="emotion-0"
    display="flex"
  >
    <div
      class="emotion-1"
      display="flex"
    />
    <div
      class="emotion-2"
      display="flex"
      height="100%"
    >
      <div
        class="emotion-3"
        display="flex"
      />
      <div
        class="emotion-4"
        display="flex"
        height="100%,100%,auto"
        width="100%,100%,672"
      >
        <div
          class="emotion-5"
          display="flex"
        />
        <div
          class="emotion-6"
          display="flex"
          width="343,343,504"
        >
          <div
            class="emotion-7"
            display="flex"
            height="100%"
          >
            <div
              class="emotion-8"
              display="flex"
            />
            <div
              class="emotion-9"
              display="flex"
            >
              <span
                class="emotion-10"
                color="text.primary"
                display="inline-flex"
                font-family="primary"
                font-size="semiBig,semiBig,big"
                font-weight="bold"
              >
                title
              </span>
            </div>
            <div
              class="emotion-11"
              display="flex"
            />
            <div
              class="emotion-9"
              display="flex"
            >
              <span
                class="emotion-13"
                color="text.gray"
                display="inline-flex"
                font-family="primary"
                font-size="default"
                font-weight="normal"
              />
            </div>
            <div
              class="emotion-14"
              display="flex"
            />
            <div
              class="emotion-15"
              display="flex"
            />
          </div>
        </div>
        <div
          class="emotion-16"
          display="flex"
        />
      </div>
      <div
        class="emotion-17"
        display="flex"
      />
    </div>
  </div>
</DocumentFragment>
`;
